core/src/main/java/com/uber/m3/tally/NoopScope.java

```diff
@@ -0,0 +1,93 @@
+package com.uber.m3.tally;
+
+import java.util.Map;
+
+import com.uber.m3.util.Duration;
+
+/**
+ * Noop implementation of Scope.
+ */
+public class NoopScope implements Scope {
+
+    @Override
+    public Counter counter(String name) {
+        return (delta) -> {
```

Comment:
this might create a new lambda object every time [ not sure it the compiler will inline this ]

Reply:
same goes for all of the other `NoOp` methods

Reply:
good point. I updated the PR.